src/mapping/attractor_mapping_recurrences.jl

```diff
@@ -501,22 +534,41 @@ function relabel_visited_cell!(bsn_nfo::BasinsInfo, old_label, new_label)
 end
 
 function basin_cell_index(y_grid_state, bsn_nfo::BasinsInfo{B}) where {B}
```

Comment:
Please split this function into dwo functions that dispatch on the `grid_nfo` field.

Write `basin_cell_index(y, bsn_nfo) = basin_cell_index(y, bsn_nfo.grid_nfo)` and then use multiple dispatch to create two functions that dispatch on the grid type.